geomstats/geometry/special_orthogonal.py

```diff
@@ -138,33 +126,72 @@ def random_uniform(self, n_samples=1, tol=1e-6):
         return self.exp(skew)
 
-    @geomstats.vectorization.decorator(['else', 'vector'])
-    def skew_matrix_from_vector(self, vec):
-        """Get the skew-symmetric matrix derived from the vector.
+    def rotation_from_angle(self, theta):
+        """Create 2D rotation matrices from scalar angles.
 
-        In nD, fill a skew-symmetric matrix with the values of the vector.
-
-        Parameters
-        ----------
-        vec : array-like, shape=[..., dim]
-            Vector.
-
-        Returns
-        -------
-        skew_mat : array-like, shape=[..., n, n]
-            Skew-symmetric matrix.
+        :param theta: float, or list of.
+        :return array-like, shape = [2, 2], or list of.
         """
-        n_vecs, vec_dim = gs.shape(vec)
-
-        mat_dim = gs.cast(
-            ((1. + gs.sqrt(1. + 8. * vec_dim)) / 2.), gs.int32)
-        skew_mat = gs.zeros((n_vecs,) + (self.n,) * 2)
-        upper_triangle_indices = gs.triu_indices(mat_dim, k=1)
-
-        for i in range(n_vecs):
-            skew_mat[i][upper_triangle_indices] = vec[i]
-            skew_mat[i] = skew_mat[i] - gs.transpose(skew_mat[i])
-        return skew_mat
+        if self.n != 2:
+            raise NotImplementedError
+        c = gs.cos(theta)
+        s = gs.sin(theta)
+        mat_first_line = gs.transpose(gs.array([c, -s]))
+        mat_second_line = gs.transpose(gs.array([s, c]))
+        mat_first_line = gs.expand_dims(
+            mat_first_line, -2)
+        mat_second_line = gs.expand_dims(mat_second_line, -2)
+        mat = gs.concatenate(
+            [mat_first_line, mat_second_line], axis=-2)
+        return mat
+
+    def angle_of_rot2(self, point):
+        """Get angle of a 2D rotation; vectorized version.
+
+        :param r: array-like (or a list of), shape=[n, n]: rotation
+        :return: float (or a list of): angle of rotation
+        """
+        if self.n != 2:
+            raise NotImplementedError
+        sinus = point[..., 1, 0]
+        cosinus = point[..., 0, 0]
+        return gs.einsum(
+            '...,...->...', gs.sign(sinus), gs.arccos(cosinus))
+
+    def multiply_angle_of_rot2(self, point, alpha):
+        """Divide the angle of a 2D rotation by a scalar; vectorized version.
+
+        Best to have original angle in :math [0, PI].
+        :param point: array-like (or a list of), shape = [n, n]: rotation
+        :param alpha: float: scalar (or a list of) to divide of angle by
+        :return: array-like (or a list of), shape=[n, n]:
+                 rotation with divided angle
+        """
+        if self.n != 2:
+            raise NotImplementedError
+        angle = gs.einsum(
+            '...,...->...', self.angle_of_rot2(point), alpha)
+        return self.rotation_from_angle(angle)
+
+    def random_gaussian(self, mean, var, n_samples=1):
+        """Emulate a Gaussian distribution in SO(n).
+
+        WARNING: This was only tested and may be viable only in SO(2).
```

Comment:
Remove the WARNING (we do not want this in the public documentation website, and the raise NotImplementedError already takes care of it).